### PR TITLE
Initial Code Refactoring of beagle-tester.c

### PR DESCRIPTION
**Pull request by lorforlinux**
@jadonk The changes form this PR will be the base for upcoming code refactoring, cleaning, and adding new code. Please review and let me know if the changes are acceptable.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# ignore binaries
+beagle-tester
```
Comment (jadonk) [inline .gitignore:2]: Makes sense. Might be useful to have some binaries place in releases. Could do more to improve release process, which mostly depends on the buildroot image builds.

File: Makefile [hunk 2]
```diff
@@ -4,12 +4,14 @@ MAKE := make
 RM := rm
 INSTALL := install
 GIT_VERSION := $(shell git describe --abbrev=6 --dirty --always --tags)
+INC_DIR = include
+# DEPS = $(INC_DIR)/common.h
```
Comment (jadonk) [inline Makefile]: If you are going to split it out, you really do need to add the dependencies to the Makefile. If it is just a handful of files, this should be easy to manage.

File: Makefile [hunk 3]
```diff
@@ -4,12 +4,14 @@ MAKE := make
...
 
 all: beagle-tester
 
 beagle-tester: beagle-tester.c
```
Comment (jadonk) [inline Makefile]: Just put $DEPS on the right hand side.

**Review by jadonk — CHANGES_REQUESTED** [review]
Direction seems OK, but please add the dependencies to Makefile as you start to split things out.

File: .gitignore (hunk 1 above)
Comment (lorforlinux) [reply .gitignore:2]: What do you suggest we should do for release process?
I was thinking as it's a binary we can just release that with an `install.sh` file?

File: Makefile (hunk 2 above)
Comment (lorforlinux) [reply Makefile]: No, I don't want to handle each dependency separately in the makefile, How to do automatic dependency management?

File: Makefile (hunk 3 above)
Comment (lorforlinux) [reply Makefile]: How do other makefiles handle dependencies without explicitly stating what files to use?

**Comment by lorforlinux** [comment]
@jadonk I have refactored some more code and edited the Makefile please review, it might not be up to the mark please guide for better code submission here.

File: Makefile (hunk 2 above)
Comment (jadonk) [reply Makefile]: This is an advanced Makefile topic and I think we are better off doing it manually for a small number of files. You can use wildcards, but they are more error prone than you imagine: https://stackoverflow.com/questions/28358957/makefile-wildcard-dependencies

Comment (jadonk) [reply Makefile]: Better, but still incomplete answer: https://www.math.colostate.edu/~yzhou/computer/writemakefile.html#Using%20Wildcards

Comment (jadonk) [reply Makefile]: grrr.... everyone recommends them: https://pages.tacc.utexas.edu/~eijkhout/istc/html/gnumake.html#Wildcards

I cannot easily find the general issues I've encountered, so I guess let's use wildcards and maybe we won't rediscover them.

File: .gitignore (hunk 1 above)
Comment (jadonk) [reply .gitignore:2]: It gets handled by debian packaging for the standard debian distro and by buildroot separately. No need for install.sh. 'make install' is good enough.

File: Makefile [hunk 4]
```diff
@@ -4,12 +4,14 @@ MAKE := make
...
 INSTALL := install
 GIT_VERSION := $(shell git describe --abbrev=6 --dirty --always --tags)
+SRC := $(wildcard src/*.c)
+CFLAGS := -O3 -W -Wall $(SRC) -Wwrite-strings -I. -I./include
 
 all: beagle-tester
 
-beagle-tester: beagle-tester.c
-	#$(CC) $(CFLAGS_FOR_BUILD) -W -Wall -Wwrite-strings -O3 -o beagle-tester beagle-tester.c -lroboticscape
-	$(CC) -DVERSION=\"${GIT_VERSION}\" $(CFLAGS_FOR_BUILD) -W -Wall -Wwrite-strings -O3 -o beagle-tester beagle-tester.c
```
Comment (jadonk) [inline Makefile:12]: CFLAGS_FOR_BUILD is a buildroot thing. It might need to come back.

File: src/beagle-tester.c [hunk 5]
```diff
@@ -280,2731 +199,9 @@ static void do_fill_screen(struct fb_info *fb_info, int pattern)
...
- * this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-
-/**********************************************/
-/*                                            */
-/*       Font file generated by cpi2fnt       */
-/*                                            */
-/**********************************************/
-
-char fontdata_8x8[] = {
```
Comment (jadonk) [inline src/beagle-tester.c:444]: I really hate when data is allocated in .h files.

File: include/font_8x8.h [hunk 6]
```diff
@@ -0,0 +1,2589 @@
...
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**********************************************/
+/*                                            */
+/*       Font file generated by cpi2fnt       */
+/*                                            */
+/**********************************************/
+
+char fontdata_8x8[] = {
```
Comment (jadonk) [inline include/font_8x8.h:26]: This should be a .c file and extern'ed in the main .c file or a .h file.

File: Makefile [hunk 7]
```diff
@@ -4,12 +4,14 @@ MAKE := make
...
+beagle-tester:
```
Comment (jadonk) [inline Makefile:12]: No dependency checks at all?

File: Makefile (hunk 4 above)
Comment (lorforlinux) [reply Makefile:12]: Ohh okay, I didn't know that.

File: src/beagle-tester.c (hunk 5 above)
Comment (lorforlinux) [reply src/beagle-tester.c:444]: I don't know it seemed obvious to me to include that in .h file.

File: include/font_8x8.h (hunk 6 above)
Comment (lorforlinux) [reply include/font_8x8.h:26]: I will fix this with next PR

File: Makefile (hunk 7 above)
Comment (lorforlinux) [reply Makefile:12]: It didn't seem necessary to do that now.

**Pull request merged**